### ISSUE TITLE
plot2d discrete with x range computes y range over all data, not restricted to x range

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2685]
Imported from SourceForge on 2024-07-06 18:07:26
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2024-02-21 02:31:29
Original: https://sourceforge.net/p/maxima/bugs/4263

---

Working w/ version 3a26747 from Git on Linux. `plot2d([discrete, ...], [x, ...])` looks at the entire data set to figure out the y range, but it should only look at the points falling in the x range.

```
(%i2) ii: makelist (i, i, 1, 100) $
```
Following plot shows the y range is 0 to 100, which is far too wide.
```
(%i3) plot2d ([discrete, ii, 100/ii], [x, 50, 100]);
(%o3)                         false
```
Following plot shows actual range of data for x from 50 to 100. `plot2d` should find that by itself.
```
(%i4) plot2d ([discrete, ii, 100/ii], [x, 50, 100], [y, 0, 2]);
(%o4)                         false
```

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:07:28
Created by **[peterpall](https://sourceforge.net/u/peterpall/)** on 2024-02-21 07:45:24
Original: https://sourceforge.net/p/maxima/bugs/4263/#4ba5

---

the same seems to be true for `draw()`.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 18:07:31
Created by **[villate](https://sourceforge.net/u/villate/)** on 2024-02-21 11:27:15
Original: https://sourceforge.net/p/maxima/bugs/4263/#b231

---

I agree with Robert. plot2d should find a suitable y range automatically. This is a remnant from the old version of plot2d where, by default, it was left to Gnuplot to decide the y-range. In a recent version of plot2d I made a change to automatically set up a y-range, a little bigger than the distance between the minimum and maximum values of y. But I did that change only for the case of an expression. The same change should be now made for the discrete-data case. Moreover, to be consistent with the case of an expression, only those points that fall within the given x range should be passed to Gnuplot (in Robert's example Gnuplot is given 100 points, while only 50 of them are shown). Sending to Gnuplot only the relevant points would make the plot look better, but in some cases when the points fall in the upper or lower bounds of the plotting box it will look even better if the y range is given to Gnuplot rather than letting it decide what to use (as we currently do in the case of an expression).
